dataInterface/movies.js

```diff
@@ -1,111 +1,234 @@
+
+  const query = {genres: { $in: [genreName] }};
+  let movieCursor = await movies.find(query).limit(10).project({title: 1, genres: 1}).sort({title: 1});
 
   return movieCursor.toArray();
 }
 
+
+// GET MOVIES BY ID
 // https://www.mongodb.com/docs/drivers/node/current/usage-examples/findOne/
 module.exports.getById = async (movieId) => {
   const database = client.db(databaseName);
   const movies = database.collection(collName);
-  const query = {_id: ObjectId(movieId)};
+
+  const query = { _id: ObjectId(movieId) };
   let movie = await movies.findOne(query);
 
   return movie;
 }
 
+// GET MOVIES BY TITLE
 module.exports.getByTitle = async (title) => {
   const database = client.db(databaseName);
   const movies = database.collection(collName);
-  const query = {title: title};
+
+  const query = { title: title };
   let movie = await movies.findOne(query);
 
   return movie;
 }
 
+// GET MOVIES BY ID OR TITLE (USES getByTitle AND getByTitle)
 module.exports.getByIdOrTitle = async (identifier) => {
   let movie;
 
-  if(ObjectId.isValid(identifier)){
+  if (ObjectId.isValid(identifier)) {
     movie = module.exports.getById(identifier);
   } else {
     movie = module.exports.getByTitle(identifier);
   }
 
-  if(movie){
+  if (movie) {
     return movie;
   } else {
-    return {error: `No item found with identifier ${identifier}.`}
+    return { error: `No item found with identifier ${identifier}.` }
   }
 }
 
+// FIND A SINGLE COMMENT FOR A MOVIE
+module.exports.getOneComment = async (commentId) => {
+  const database = client.db(databaseName);
+  const comments = database.collection(commCollName);
+
+  const query = { _id: ObjectId(commentId) };
+  let comment = await comments.findOne(query);
+
+  return comment;
+}
+
+// CREATE A NEW MOVIE
 // https://www.mongodb.com/docs/v4.4/tutorial/insert-documents/
 module.exports.create = async (newObj) => {
   const database = client.db(databaseName);
   const movies = database.collection(collName);
 
-  if(!newObj.title){
+  if (!newObj.title) {
     // Invalid movie object, shouldn't go in database.
-    return {error: "Movies must have a title."}
+    return { error: "Movies must have a title." }
   }
   const result = await movies.insertOne(newObj);
 
-  if(result.acknowledged){
+  if (result.acknowledged) {
     return { newObjectId: result.insertedId, message: `Item created! ID: ${result.insertedId}` }
+  } else {
+    return { error: "Something went wrong. Please try again." }
+  }
+}
+
+// function isValidObjectId(id){
+// if(ObjectId.isValid(id)){
+//   if((String)(new ObjectId(id)) === id)
+//       return true;       
+//   return false;
+// }
+// return false;
+// }
+
+// CREATE A NEW COMMENT
+module.exports.createComment = async (movieId, newObj) => {
+  // TODO: Validate that movieId is for an existing movie
+  const database = client.db(databaseName);
+  const comments = database.collection(commCollName);
+  const validMovie = await module.exports.getById(movieId);
```

Comment:
This is a good check, but it would be more valuable to test if it exists before inserting the new comment (line 141), rather than just testing when sending an error (line 146).